tfe/provider.go

```diff
@@ -0,0 +1,200 @@
+			"tfe_sentinel_policy":    resourceTFESentinelPolicy(),
+			"tfe_ssh_key":            resourceTFESSHKey(),
+			"tfe_team":               resourceTFETeam(),
+			"tfe_team_access":        resourceTFETeamAccess(),
+			"tfe_team_member":        resourceTFETeamMember(),
+			"tfe_team_members":       resourceTFETeamMembers(),
+			"tfe_team_token":         resourceTFETeamToken(),
+			"tfe_workspace":          resourceTFEWorkspace(),
+			"tfe_variable":           resourceTFEVariable(),
+		},
+
+		ConfigureFunc: providerConfigure,
+	}
+}
+
+func providerConfigure(d *schema.ResourceData) (interface{}, error) {
+	// Get the hostname and token.
+	hostname := d.Get("hostname").(string)
+	token := d.Get("token").(string)
+
+	// Parse the hostname for comparison,
+	host, err := svchost.ForComparison(hostname)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the Terraform CLI configuration.
+	config := cliConfig()
+
+	// Create a new credential source and service discovery object.
+	credsSrc := credentialsSource(config)
+	services := disco.NewWithCredentialsSource(credsSrc)
+
+	// Add any static host configurations service discovery object.
+	for userHost, hostConfig := range config.Hosts {
+		host, err := svchost.ForComparison(userHost)
+		if err != nil {
+			// ignore invalid hostnames.
+			continue
+		}
+		services.ForceHostServices(host, hostConfig.Services)
+	}
+
+	// Discover the full Terraform Enterprise service address.
+	address := services.DiscoverServiceURL(host, serviceID)
+	if address == nil {
+		return nil, fmt.Errorf("host %s does not provide a Terraform Enterprise API", host)
+	}
+
+	// Only try to get to the token from the credentials source if no token
+	// was explicitly set in the provider configuration.
+	if token == "" {
+		creds, err := services.CredentialsForHost(host)
+		if err != nil {
+			log.Printf("[DEBUG] Failed to get credentials for %s: %s (ignoring)", host, err)
+		}
+		if creds != nil {
+			token = creds.Token()
+		}
+	}
+
+	// If we still don't have a token at this point, we return an error.
+	if token == "" {
+		return nil, fmt.Errorf("required token could not be found")
+	}
+
+	// Create a new TFE client config..
+	cfg := &tfe.Config{
+		Address: address.String(),
+		Token:   token,
+	}
+
+	// Create s new TFE client.
+	return tfe.NewClient(cfg)
+}
+
+// cliConfig tries to find and parse the configuration of the Terraform CLI.
+// This is an optional step, so any errors are ignored.
+func cliConfig() *Config {
+	config := &Config{}
+
+	// Detect the CLI config file path.
+	configFilePath := os.Getenv("TERRAFORM_CONFIG")
+	if configFilePath == "" {
+		filePath, err := configFile()
+		if err != nil {
+			log.Printf("[ERROR] Error detecting default CLI config file path: %s", err)
+			return config
+		}
+		configFilePath = filePath
+	}
+
+	// Read the CLI config file content.
+	content, err := ioutil.ReadFile(configFilePath)
+	if err != nil {
+		log.Printf("[ERROR] Error reading the CLI config file: %v", err)
```

Comment:
Including the config path in these will really help with debugging.

Reply:
Will fix that in a bit...

Reply:
Fixed